x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapperTests.java

```diff
@@ -581,7 +604,11 @@ public void testSuccessfulParse() throws IOException {
             assertNull(luceneDocs.get(3).getParent());
 
             withLuceneIndex(mapperService, iw -> iw.addDocuments(doc.docs()), reader -> {
-                NestedDocuments nested = new NestedDocuments(mapperService.mappingLookup(), QueryBitSetProducer::new, indexVersion);
+                NestedDocuments nested = new NestedDocuments(
+                    mapperService.mappingLookup(),
+                    QueryBitSetProducer::new,
+                    IndexVersion.current()
```

Comment:
Better to use `getVersion()` here in case we override the version

Reply:
We don't override the version here since we rely on the fact that the version should work with the legacy and new format. It's also only related to nested documents and the check for the index version is very old and not relevant here,